packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -30,13 +32,26 @@ export namespace TerminalCommands {
     };
 }
 
+const TERMINAL_ACTIVE_CONTEXT = "terminalActive";
+
+/* Return true if the event targets xterm's internal textarea.  */
+
+function isTerminalFocused(binding: Keybinding, event: KeyboardEvent): boolean {
+    if (event.target instanceof HTMLTextAreaElement) {
```

Comment:
You should be able to check it with `shell.currentWidget instanceof TerminalWidget` instead of exposing `KeyboardEvent`.

Reply:
How do I get a reference to `shell`?

Reply:
Ahh, through `this.app.shell`.